pkg/monitor/monitor.go

```diff
@@ -119,44 +126,54 @@ func Monitor(clusterName, clusterDomain, templatePath, cfgPath, apiVip string, a
 			if err != nil {
 				return err
 			}
-			newConfig = &config
-			if oldConfig == nil || !cmp.Equal(*oldConfig, *newConfig) {
-				log.WithFields(logrus.Fields{
-					"newConfig": *newConfig,
-				}).Info("Config change detected")
-				err = render.RenderFile(cfgPath, templatePath, RuntimeConfig{LBConfig: newConfig})
-				if err != nil {
-					log.WithFields(logrus.Fields{
-						"config": *newConfig,
-					}).Error("Failed to render HAProxy configuration")
-					return err
-				}
-				_, err = conn.Write([]byte("reload\n"))
-				if err != nil {
-					log.WithFields(logrus.Fields{
-						"socket": haproxyMasterSock,
-					}).Error("Failed to write reload to HAProxy master socket")
-					return err
-				}
+			curConfig = &config
+			if appliedConfig == nil || !cmp.Equal(*appliedConfig, *curConfig) {
+			    if prevConfig == nil || cmp.Equal(*prevConfig, *curConfig) {configChangeCtr++}else {configChangeCtr=1}
+			    log.WithFields(logrus.Fields{
+			     "curConfig": *curConfig,
+			     "configChangeCtr": configChangeCtr,
+			    }).Info("Config change detected")
+			    if configChangeCtr >= cfgChangeThreshold {
+			        log.WithFields(logrus.Fields{
+			         "curConfig": *curConfig,
+			        }).Info("Apply config change")
+			        err = render.RenderFile(cfgPath, templatePath, RuntimeConfig{LBConfig: curConfig})
+			        if err != nil {
+			            log.WithFields(logrus.Fields{
+			             "config": *curConfig,
+			            }).Error("Failed to render HAProxy configuration")
+			            return err
+			        }
+			        _, err = conn.Write([]byte("reload\n"))
+			        if err != nil {
+			            log.WithFields(logrus.Fields{
+			             "socket": haproxyMasterSock,
+			             }).Error("Failed to write reload to HAProxy master socket")
+			             return err
+			        }
+			        configChangeCtr = 0
+			        appliedConfig = curConfig
+			    }
+			}else {
+			    configChangeCtr=0
 			}
-			oldConfig = newConfig
-
-			ok, err := utils.IsKubernetesHealthy(lbPort)
-			if err == nil && ok {
-				if !k8sIsHealthy {
-					log.Info("API is reachable through HAProxy")
-					k8sIsHealthy = true
-				}
-				err := ensureHAProxyPreRoutingRule(apiVip, apiPort, lbPort)
-				if err != nil {
-					log.WithFields(logrus.Fields{"err": err}).Error("Failed to ensure HAProxy PREROUTING rule to direct traffic to the LB")
-				}
-			} else {
-				cleanHAProxyPreRoutingRule(apiVip, apiPort, lbPort)
-				if k8sIsHealthy {
-					log.Info("API is not reachable through HAProxy")
-					k8sIsHealthy = false
-				}
+			prevConfig = &config
+
+			curK8sHealthkSts, err := utils.IsKubernetesHealthy(lbPort)
+			if err != nil {curK8sHealthkSts=false}
```

Comment:
what does `curK8sHealthkSts` stand for?

Reply:
The idea is to have stabilization for k8s health status (for both on and off), so  'curK8sHealthkSts' holds the value of single (latest) k8s health check and 'K8sHealthSts' holds the value after stabilization.

The only usage of 'curK8sHealthkSts' is a parameter to stabilization function, and is being calculated as follows:

'curK8sHealthkSts, err := utils.IsKubernetesHealthy(lbPort)
if err != nil {curK8sHealthkSts=false}
'
I thought at the beginning to pass the output of 'utils.IsKubernetesHealthy(lbPort)'  as a parameter to the stabilization function (and not define a local variable for that case), but I don't know how to handle the case where err != NIL (new to GOLANG)

If you think there's a way to avoid using a local variable for that purpose, I'll be glad to change it.

Reply:
Is Healthk a typo?

Reply:
Yep, 10x!